janMarkusMilan/assignments/treesBFS/lc102/lc102.js

```diff
@@ -1,24 +1,99 @@
 // Problem Statement #
 
-// Given a binary tree, populate an array to represent its level-by-level traversal. You should populate the values of all nodes of each level from left to right in separate sub-arrays.
+// Given a binary tree, populate an array to represent its level-by-level traversal.
+// You should populate the values of all nodes of each level from left to right in separate sub-arrays.
 
-class TreeNode {
+/*
+Answer Format: [[1], [2, 3], [4, 5, 6]]
+Edge Case: !root -> []
+
+Implement BFS for our answer
+
+function BFS (root) {
+  res = [];
+  q = [root]
+
+  while (q.length) {
+    node = q.shift();
+    res.push(node.val);
+
+    if (node.left) {
+      q.push(node.left);
+    }
+
+    if (node.right) {
+      q.push(node.right)
+    }
+  }
+
+  return res;
+}
+
+[1, 2, 3, 4, 5, 6]
 
+[1, 2, 3]
+node = [1]
+res = [1]
+
+    (1)
+    / \
+  (2) (3)
+  / \
+(4) (5)
+
+res = [1]
+q = [1, 2, 3]
+
+Initialize a new array in our loop
+Once we reach the end of a level, push sub-array into our res array
+
+Keep track of the length of the queue
+Push into a sub-array
+Push once the queue reaches a certain breakpoint
+*/
+
```

Comment:
Great job on this approach! It may help if you write out in sentences/phrases what your code is going to do instead of writing out code here. But you explained it very well during mock technicals.